RecoEcal/EgammaCoreTools/src/EcalClusterLazyTools.cc

```diff
@@ -257,11 +257,11 @@ float EcalClusterLazyToolsBase::eseffsirir(const reco::SuperCluster &cluster)
   if (!(fabs(cluster.eta()) > 1.6 && fabs(cluster.eta()) < 3.)) return 0.;
 
   const CaloSubdetectorGeometry *geometryES = geometry_->getSubdetectorGeometry(DetId::Ecal, EcalPreshower);
-  CaloSubdetectorTopology *topology_p = 0;
-  if (geometryES) topology_p = new EcalPreshowerTopology(geometry_);
+  std::unique_ptr<CaloSubdetectorTopology> topology_p;
+  if (geometryES) topology_p.reset(new EcalPreshowerTopology(geometry_));
 
-  std::vector<float> phoESHitsIXIX = getESHits(cluster.x(), cluster.y(), cluster.z(), rechits_map_, geometry_, topology_p, 0, 1);
-  std::vector<float> phoESHitsIYIY = getESHits(cluster.x(), cluster.y(), cluster.z(), rechits_map_, geometry_, topology_p, 0, 2);
+  std::vector<float> phoESHitsIXIX = getESHits(cluster.x(), cluster.y(), cluster.z(), rechits_map_, geometry_, topology_p.get(), 0, 1);
+  std::vector<float> phoESHitsIYIY = getESHits(cluster.x(), cluster.y(), cluster.z(), rechits_map_, geometry_, topology_p.get(), 0, 2);
```

Comment:
@jolange: Will there be any cases where this code is called with the ECAL preshower missing from the geometry? Maybe some kind of upgrade testing, or processing cosmics, or other specialized testing? If you put in the assert, then the program will not be able to run without the ECAL preshower, even though it might not actually be otherwise needed. For maximum flexibility, it would probably be better to preserve the current behavior and make sure the case of a null `geometryES` is handled correctly.